drivers/include/ng_at86rf2xx.h

```diff
@@ -0,0 +1,359 @@
+ *
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef NG_AT86RF2XX_H_
+#define NG_AT86RF2XX_H_
+
+#include <stdint.h>
+
+#include "board.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "net/ng_netdev.h"
+#include "ng_at86rf2xx.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Maximum possible packet size in byte
+ */
+#define NG_AT86RF2XX_MAX_PKT_LENGTH     (127)
+
+/**
+ * @brief   Default addresses used if the CPUID module is not present
+ * @{
+ */
+#define NG_AT86RF2XX_DEFAULT_ADDR_SHORT (0x0230)
+#define NG_AT86RF2XX_DEFAULT_ADDR_LONG  (0x1122334455667788)
+/** @} */
+
+/**
+  * @brief   Channel configuration
+  * @{
+  */
+#ifdef MODULE_NG_AT86RF212B
+/* the AT86RF212B has a sub-1GHz radio */
+#define NG_AT86RF2XX_MIN_CHANNEL        (0)
+#define NG_AT86RF2XX_MAX_CHANNEL        (10)
+#define NG_AT86RF2XX_DEFAULT_CHANNEL    (5)
+#else
+#define NG_AT86RF2XX_MIN_CHANNEL        (11U)
+#define NG_AT86RF2XX_MAX_CHANNEL        (26U)
+#define NG_AT86RF2XX_DEFAULT_CHANNEL    (17U)
+#endif
+/** @} */
+
+/**
+ * @brief   Default PAN ID
+ *
+ * TODO: Read some global network stack specific configuration value
+ */
+#define NG_AT86RF2XX_DEFAULT_PANID      (0x0023)
+
+/**
+ * @brief   Default TX power (0dBm)
+ */
+#define NG_AT86RF2XX_DEFAULT_TXPOWER    (0U)
+
+/**
+ * @brief   Device internal states (see datasheet)
+ */
+typedef enum {
+    STATE_TRX_OFF       = 0x08,     /**< idle */
+    STATE_PLL_ON        = 0x09,     /**< ready to transmit */
+    STATE_SLEEP         = 0x0f,     /**< sleep mode */
+    STATE_BUSY_RX_AACK  = 0x11,     /**< busy receiving data */
+    STATE_BUSY_TX_ARET  = 0x12,     /**< busy transmitting data */
+    STATE_RX_AACK_ON    = 0x16,     /**< waiting for incoming data */
+    STATE_TX_ARET_ON    = 0x19,     /**< ready for sending data */
+    STATE_IN_PROGRESS   = 0x1f,     /**< state conversion in progress */
+} ng_at86rf2xx_state_t;
+
+/**
+ * @brief   Internal device options
+ */
+typedef enum {
+    OPT_AUTOACK         = 0x0001,   /**< auto ACKs active */
+    OPT_CSMA            = 0x0002,   /**< CSMA active */
+    OPT_PROMISCUOUS     = 0x0004,   /**< promiscuous mode active */
+    OPT_PRELOADING      = 0x0008,   /**< preloading enabled */
+    OPT_TELL_TX_START   = 0x0010,   /**< notify MAC layer on TX start */
+    OPT_TELL_TX_END     = 0x0020,   /**< notify MAC layer on TX finished */
+    OPT_TELL_RX_START   = 0x0040,   /**< notify MAC layer on RX start */
+    OPT_TELL_RX_END     = 0x0080,   /**< notify MAC layer on RX finished */
+    OPT_RAWDUMP         = 0x0100,   /**< pass RAW frame data to upper layer */
+    OPT_SRC_ADDR_LONG   = 0x0200,   /**< send data using long source address */
+    OPT_USE_SRC_PAN     = 0x0400,   /**< do not compress source PAN ID */
+} ng_at86rf2xx_opt_t;
```

Comment:
Please see https://github.com/RIOT-OS/RIOT/pull/2614#discussion_r28945055 for argumentation against `enum`-based flag definitions.